src/js/components/Table.js

```diff
@@ -120,35 +133,64 @@ export default class Table extends Component {
     }
   }
 
+  _adjustBodyCells () {
+    //adjust table body cells to have link to the header
+    //so that in responsive mode it displays the text as content in css
+    let headerCells = this.refs.table.querySelectorAll('thead th');
+    if (headerCells.length > 0) {
+      let rows = this.refs.table.querySelectorAll('tbody tr');
+
+      [].forEach.call(rows, (row) => {
+        [].forEach.call(row.cells, (cell, index) => {
+          cell.setAttribute('data-th', headerCells[index].innerText);
+        });
+      });
+    }
+  }
+
   _onResize () {
     this._alignMirror();
+    this._layout();
   }
 
-  _buildMirror () {
-    var tableElement = this.refs.table;
-    var cells = tableElement.querySelectorAll('thead tr th');
-    var mirrorElement = this.refs.mirror;
-    var mirrorRow = mirrorElement.querySelectorAll('thead tr')[0];
-    while (mirrorRow.hasChildNodes()) {
-      mirrorRow.removeChild(mirrorRow.lastChild);
+  _layout () {
+    let availableSize = this.refs.container.offsetWidth;
+    let numberOfCells = this.refs.table.querySelectorAll('thead th').length;
+
+    if ((numberOfCells * MIN_CELL_WIDTH) > availableSize) {
```

Comment:
Add a comment explaining the strategy here. It does have some limitations depending on how rich the data is in the table cells.

One other idea is to check for the header mirror overflowing the table's container and switch to small mode in that case. This would work OK for shrinking but might cause flicker when expanding. Something that uses the actual content to drive behavior would be preferable if possible.